src/express-tooling.ts

```diff
@@ -105,27 +114,39 @@ function handleOAuthRequestMiddleware(options: MiddlewareOptions): RequestHandle
     const authHeader = getHeaderValue(req, AUTHORIZATION_HEADER_FIELD_NAME);
 
     if (!authHeader) {
+      if (logger) {
+        logger.error('No authorization field in header');
+      }
       rejectRequest(res);
       return;
     }
 
     const accessToken = extractAccessToken(authHeader);
     if (!accessToken) {
+      if (logger) {
+        logger.error('No access_token field in header');
+      }
       rejectRequest(res);
       return;
     } else {
       getTokenInfo(tokenInfoEndpoint, accessToken)
-      .then(setTokeninfo(req))
-      .then(next)
-      .catch(err => rejectRequest(res, err.status));
+        .then(setTokeninfo(req))
+        .then(next)
+        .catch(err => {
+          if (logger) {
+            logger.error('Error occured when requesting token info: ', err);
+          }
+          return rejectRequest(res, err.status);
+        });
     }
   };
 }
 
 function validateScopes(req: ExtendedRequest,
                         res: Response,
                         next: NextFunction,
-                        scopes: string[] = []): void {
+                        scopes: string[] = [],
```

Comment:
I am not a fan of multiple optional paramters / default values. 
We could remove here the default value, and make sure that the caller passed in the right value.

Reply:
👍